Simulator/libSimulator/DOT.h

```diff
@@ -32,13 +32,17 @@ class DOT : public Debuff, public Ability {
             return std::nullopt;
         return _lastTickTime + _tickRate;
     }
-
+    void setPlayer(PlayerPtr player) {_player= player;}
```

Comment:
making the DOT have information about who actually applied it (a shared pointer in this case so it can refer back to it)